app/src/main/java/com/android/sample/ui/leaderboard/LeaderboardScreen.kt

```
@@ -0,0 +1,532 @@
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.android.sample.model.leaderboard.LeaderboardCache
+import com.android.sample.model.profile.UserProfile
+import com.android.sample.ui.leaderboard.LeaderboardAddOns.crown
+import com.android.sample.ui.leaderboard.LeaderboardAddOns.cutiePatootie
+import com.android.sample.ui.leaderboard.LeaderboardBadgeThemes.CutieColor
+import com.android.sample.ui.leaderboard.LeaderboardBadgeThemes.forRank
+import com.android.sample.ui.navigation.BottomNavigationMenu
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.NavigationTab
+import com.android.sample.ui.navigation.Screen
+import com.android.sample.ui.navigation.TopNavigationBar
+import com.android.sample.ui.profile.ProfilePicture
+import com.android.sample.ui.profile.UserSections
+import com.android.sample.ui.theme.appPalette
+import com.android.sample.ui.utils.EnumFilterButton
+import com.android.sample.ui.utils.EnumFilterPanel
+import com.android.sample.ui.utils.RangeFilterButton
+import com.android.sample.ui.utils.RangeFilterPanel
+
+/** Top-level screen composable for the leaderboard. */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun LeaderboardScreen(
+    modifier: Modifier = Modifier,
+    leaderboardViewModel: LeaderboardViewModel? = null,
+    searchFilterViewModel: LeaderboardSearchFilterViewModel = viewModel(),
+    navigationActions: NavigationActions? = null,
```

Used in TopNavigationBar and BottomNavigationMenu without null checks. If null, clicks will silently fail or crash.